src/Datastore/EntityMapper.php

```diff
@@ -429,4 +446,18 @@ private function isEncoded($value)
 
         return true;
     }
+
+    /**
+     * Get the value type from a value object.
+     *
+     * @param array $value
+     * @return string
+     */
+    private function getValueType(array $value)
+    {
+        $keys = array_keys($value);
+        return array_values(array_filter($keys, function ($key) {
+            return in_array($key, $this->valueTypes);
+        }))[0];
```

Comment:
When it doesn't find any keys that matches the $valueTypes, it will just return null with E_NOTICE. I think it should be a WARNING or severer situation. WDYT?